src/imp/openssl.rs

```diff
@@ -161,6 +160,19 @@ impl Identity {
             chain: parsed.chain.into_iter().flatten().collect(),
         })
     }
+
+    pub fn from_pkcs8(buf: &[u8], key: &[u8]) -> Result<Identity, Error> {
+        let pkey = PKey::private_key_from_pem(key)?;
+        let mut cert_chain = X509::stack_from_pem(buf)?.into_iter();
+        let cert = cert_chain.next();
+        let chain = cert_chain.collect();
```

Comment:
Isn't this collecting a vec directly back into a vec?

Reply:
It's collecting the rest of the chain.